corehq/apps/app_manager/suite_xml.py

```diff
@@ -1950,33 +1965,25 @@ def get_target_module(case_type, module_id, with_product_details=False):
                     'action': action
                 })
             else:
-                if action.parent_tag:
-                    parent_action = form.actions.actions_meta_by_tag[action.parent_tag]['action']
+                if action.case_index.tag:
+                    parent_action = form.actions.actions_meta_by_tag[action.case_index.tag]['action']
                     parent_filter = SuiteGenerator.get_parent_filter(
-                        action.parent_reference_id,
+                        action.case_index.reference_id,
                         parent_action.case_session_var
                     )
+                    datums.append({
+                        'datum': get_manual_datum(action, parent_filter),
+                        'case_type': action.case_type,
+                        'requires_selection': True,
+                        'action': action
+                    })
                 else:
-                    parent_filter = ''
-
-                target_module = get_target_module(action.case_type, action.details_module)
-                referenced_by = form.actions.actions_meta_by_parent_tag.get(action.case_tag)
-                datum = SessionDatum(
-                    id=action.case_session_var,
-                    nodeset=(SuiteGenerator.get_nodeset_xpath(action.case_type, target_module, True) + parent_filter),
-                    value="./@case_id",
-                    detail_select=self.get_detail_id_safe(target_module, 'case_short'),
-                    detail_confirm=(
-                        self.get_detail_id_safe(target_module, 'case_long')
-                        if not referenced_by or referenced_by['type'] != 'load' else None
-                    )
-                )
-                datums.append({
-                    'datum': datum,
-                    'case_type': action.case_type,
-                    'requires_selection': True,
-                    'action': action
-                })
+                    datums.append({
+                        'datum': get_manual_datum(action),
+                        'case_type': action.case_type,
+                        'requires_selection': True,
+                        'action': action
+                    })
```

Comment:
This change doesn't seem like an improvement. I think originally you changed it since you were looping through the indices but now there is just one and so we can revert to the previous logic where the if/else just determines what the parent_filter value should be.